C7Engine/EntryPoints/MessageToEngine.cs

```diff
@@ -159,8 +159,17 @@ public override void process()
 	public class MsgEndTurn : MessageToEngine {
 		public override void process()
```

Comment:
Would it make sense to have process call out to an OnEndTurn method in the turn handler? Organisationally it might be nice to have the begin / end / advance logic all in the same place

Reply:
I don't think it would make sense to simply move that code into a method called OnEndTurn. For one thing it applies only to the UI controlling player, unlike OnBeginTurn, so they wouldn't be symmetric. For another thing, that code doesn't actually end the turn from the perspective of the engine, it only ends the movement phase for the UI controller. I plan to refactor AdvanceTurn again at some point using "yield return" to create a state machine instead of requiring the engine to call it repeatedly. Once that's done it will be easy to insert OnBegin/EndMovementPhase methods that are called for each player. Then that code in MsgEndTurn.process could be (and should be) moved into OnEndMovementPhase.

Reply:
Ok, makes sense that it's at the controller level. Sounds good to me